packages/erk-shared/src/erk_shared/gateway/graphite/abc.py

```
@@ -253,6 +253,48 @@ def get_child_branches(self, git_ops: Git, repo_root: Path, branch: str) -> list
             return []
         return all_branches[branch].children
 
+    def find_ancestor_worktree(
+        self,
+        git_ops: Git,
+        repo_root: Path,
+        branch: str,
+    ) -> WorktreeInfo | None:
+        """Find the worktree of the closest ancestor branch.
+
+        Walks up the Graphite parent chain to find the first ancestor
+        that has a worktree checked out. This is a composing template method
+        that uses get_parent_branch() to traverse the branch hierarchy.
+
+        Args:
+            git_ops: Git instance for worktree listing and git common directory
+            repo_root: Repository root directory
+            branch: Branch name to find ancestor worktree for
+
+        Returns:
+            WorktreeInfo of the closest ancestor with a worktree, or None if:
+            - Branch is not tracked by Graphite
+            - No ancestor has a worktree
+        """
+        # Validate branch exists in Graphite
+        all_branches = self.get_all_branches(git_ops, repo_root)
+        if branch not in all_branches:
+            return None
```

I think this should be an error, not just silently return None